### DISCUSSION TITLE
Structure for BTP docs section

### DISCUSSION THREAD

**Discussion by bwhli** [icon-project/devportal#42] [Ideas]
# Interoperability

### BTP Overview

* Overview of BTP technology

### Arbitrary Call Service (xCall)

* How to Perform an xCall
* Some examples for how to do common cross-chain calls

### Supported Networks
* Table of supported chains with Chain IDs
* Supported assets for free

### Reference

* BTP Litepaper
* FAQs
* Brand Assets (BTP logo)

**Comment by bwhli** [comment]
@han-so1omon For the xCall demonstration, should we provide examples to/from each network?

**Reply by han-so1omon** [reply]
I would only support documenting features that are available. At current this would be the theoretical stuff and some of the specs. Once that stuff is available, then I think very small in-place examples would be good

**Reply by CyrusVorwald** [reply]
We can reference the IIP here https://github.com/icon-project/IIPs/blob/master/IIPS/iip-52.md

**Comment by han-so1omon** [comment]
What is the "brand assets" section?

**Reply by bwhli** [reply]
Logos for BTP and ICON Bridge, but actually I don't think we need this anymore because I updated the brand kit on icon.community with BTP logos.

**Comment by han-so1omon** [comment]
I would rearrange the structure so that informational sections are separated from example sections:

Comments:
- Remove BTP sub-section, which is listed in your structure as an umbrella sub-section
- Remove "Brand Assets", unless it is necessary. I don't know what this means
- Potentially move FAQs to the "Advanced Topics" category. We should also think about what those questions are first

Questions:
- What is meant by "supported assets for free"?

Proposed rearranged structure:
- Getting Started
- - How to use the Arbitrary Call Service for blockchain interoperability => Add this section to the getting started section
- Interoperability
- - Blockchain Transmission Protocol (BTP) => This is an overview
- - - Decentralized versus Centralized => This is decentralized relays (with verifier) versus centralized relays
- - - Litepaper => This links to the pdf litepaper
- - POP Bridge => Documentation of POP Bridge product, including links to bridge webapp
- - Arbitrary Call Service (xCall) => basically a specification of the service
- - Supported Networks => List the supported networks

**Reply by rhizome-labs** [reply]
@han-so1omon Based on what we discussed on our last call, I thought this BTP and ICON Bridge would be structured under an "Interoperability" section (e.g. in the screenshot below, replace "CONCEPTS" with "INTEROPERABILITY" and "BTP" and "ICON Bridge" would be sub-sections under that).

![CleanShot 2022-07-27 at 11 59 08@2x](https://user-images.githubusercontent.com/53635700/181294174-e91bee3e-be61-42b0-9d37-8f3552403842.jpg)

How about something like this? I think this is a bit clearer because it separates BTP and ICON Bridge. I think this separation is advantageous because the two products are fundamentally different, and may diverge more and more down the line. This also allows us to list supported networks in a clearer/hierarchical fashion because the two services may support different networks.

```
# INTEROPERABILITY

## Solutions – discuss the difference between BTP and ICON Bridge (decentralized vs. centralized)

## Arbitrary Call Service (xCall) - outline the xCall spec

## BTP

### Overview - if necessary, provide detailed overview of BTP

### Supported Networks

### Resources - link to stuff like litepaper, tutorials, etc.

...other headings related to BTP

## ICON Bridge

### Overview

### Nexus

#### How to Add a Token to Nexus

#### How to Make a Cross-Chain Transaction With Nexus

### Supported Networks

## FAQ
```

What do you think about this?

**Reply by rhizome-labs** [reply]
> I would rearrange the structure so that informational sections are separated from example sections:
> 
> Comments:
> 
> * Remove BTP sub-section, which is listed in your structure as an umbrella sub-section
> * Remove "Brand Assets", unless it is necessary. I don't know what this means
> * Potentially move FAQs to the "Advanced Topics" category. We should also think about what those questions are first
> 
> Questions:
> 
> * What is meant by "supported assets for free"?
> 
> Proposed rearranged structure:
> 
> * Getting Started
> * * How to use the Arbitrary Call Service for blockchain interoperability => Add this section to the getting started section
> * Interoperability
> * * Blockchain Transmission Protocol (BTP) => This is an overview
> * * * Decentralized versus Centralized => This is decentralized relays (with verifier) versus centralized relays
> * * * Litepaper => This links to the pdf litepaper
> * * POP Bridge => Documentation of POP Bridge product, including links to bridge webapp
> * * Arbitrary Call Service (xCall) => basically a specification of the service
> * * Supported Networks => List the supported networks

I meant to say "supported assets for fees", as in what assets can be used for fees on each network. I saw this section on other interoperability solution docs, so wanted to bring it up here. On ICON, the fee will be paid in ICX. Let's say someone is originating a BTP tx from BSC, that fee would be paid in BNB, correct?

**Reply by han-so1omon** [reply]
Just to keep this thread up-to-date, BTP is getting its own docs. The format looks generally good, though
